standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/MetaStoreDirectSql.java

```diff
@@ -449,6 +449,53 @@ public Database getDatabase(String catName, String dbName) throws MetaException{
     }
   }
 
+  public List<String> getDatabases(String catName, String pattern) throws MetaException {
+    List<String> result = new ArrayList<>();
+    StringBuilder queryText = new StringBuilder("SELECT \"NAME\" FROM ")
+            .append(DBS)
+            .append(" WHERE \"CTLG_NAME\" = ?");
+
+    List<Object> params = new ArrayList<>();
+    params.add(catName);
+
+    if (pattern != null && !pattern.equals("*")) {
+      List<String> validPatterns = Arrays.stream(pattern.split("\\|"))
+              .map(String::trim)
+              .filter(s -> !s.isEmpty())
+              .distinct()
+              .map(String::toLowerCase)
+              .toList();
+      if (!validPatterns.isEmpty()) {
+        queryText.append(" AND (");
+        for (int i = 0; i < validPatterns.size(); i++) {
+          if (i > 0) queryText.append(" OR ");
+          queryText.append("\"NAME\" LIKE ?");
+          params.add(validPatterns.get(i).replace('*', '%'));
+        }
```

Comment:
Does this loop retain the original semantics? I can see the original implementation assumes `pattern` is a regular expression, which is not identical to that of SQL's `LIKE`.
https://github.com/apache/hive/blob/411d3a49cb7a2720907cf0342f5f77f53dbfed3d/standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ObjectStore.java#L2027-L2051